invenio_app_rdm/communities_ui/views/communities.py

```diff
@@ -89,3 +92,25 @@ def communities_home(pid_value, community, community_ui):
             permissions=permissions,
             records=records_ui,
         )
+
+
+@pass_community(serialize=True)
+def community_static_page(pid_value, community, community_ui):
+    """Community static page."""
+    permissions = community.has_permissions_to(
+        ["update", "read", "search_requests", "search_invites", "moderate"]
+    )
+
+    try:
+        page = current_pages_service.read_by_url(g.identity, request.path).to_dict()
+
+    except (PageNotFoundError, NoResultFound):
```

Comment:
`NoResultFound` is already being handled by the pages model [here](https://github.com/inveniosoftware/invenio-pages/blob/master/invenio_pages/records/models.py#L67). It throws a `PageNoutFoundError` that you can then catch.